Latest build seems to require Java 8

When building the latest from master, I see an error in the stacktrace indicating that it's looking for Java 8 classes:

```
Caused by: org.gradle.api.GradleException: Could not generate a proxy class for class com.google.protobuf.gradle.ProtobufSourceDirectorySet.
	at org.gradle.api.internal.AbstractClassGenerator.generateUnderLock(AbstractClassGenerator.java:204)
	at org.gradle.api.internal.AbstractClassGenerator.generate(AbstractClassGenerator.java:67)
	at org.gradle.api.internal.ClassGeneratorBackedInstantiator.newInstance(ClassGeneratorBackedInstantiator.java:36)
	at org.gradle.api.internal.plugins.DefaultConvention.create(DefaultConvention.java:106)
	at com.google.protobuf.gradle.ProtobufPlugin$_addSourceSetExtensions_closure4.doCall(ProtobufPlugin.groovy:124)
	at org.gradle.api.internal.ClosureBackedAction.execute(ClosureBackedAction.java:67)
	at org.gradle.api.internal.DefaultDomainObjectCollection.all(DefaultDomainObjectCollection.java:110)
	at org.gradle.api.internal.DefaultDomainObjectCollection.all(DefaultDomainObjectCollection.java:115)
	at org.gradle.api.DomainObjectCollection$all$1.call(Unknown Source)
	at com.google.protobuf.gradle.ProtobufPlugin.addSourceSetExtensions(ProtobufPlugin.groovy:123)
	at com.google.protobuf.gradle.ProtobufPlugin.apply(ProtobufPlugin.groovy:82)
	at com.google.protobuf.gradle.ProtobufPlugin.apply(ProtobufPlugin.groovy)
	at org.gradle.api.internal.plugins.ImperativeOnlyPluginApplicator.applyImperative(ImperativeOnlyPluginApplicator.java:35)
	at org.gradle.api.internal.plugins.RulesCapablePluginApplicator.applyImperative(RulesCapablePluginApplicator.java:42)
	at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:133)
	... 59 more
Caused by: java.lang.NoClassDefFoundError: java/util/Spliterator
	at org.gradle.api.internal.AbstractClassGenerator.inspectType(AbstractClassGenerator.java:265)
	at org.gradle.api.internal.AbstractClassGenerator.inspectType(AbstractClassGenerator.java:220)
	at org.gradle.api.internal.AbstractClassGenerator.generateUnderLock(AbstractClassGenerator.java:98)
	... 73 more
```

--- a/build.gradle
+++ b/build.gradle
@@ -130,7 +130,7 @@ subprojects {
                 okhttp: 'com.squareup.okhttp:okhttp:2.4.0',
                 protobuf: "com.google.protobuf:protobuf-java:${protobufVersion}",
                 protobuf_nano: "com.google.protobuf.nano:protobuf-javanano:${protobufVersion}",
-                protobuf_plugin: 'com.google.protobuf:protobuf-gradle-plugin:0.6.1',
+                protobuf_plugin: 'com.google.protobuf:protobuf-gradle-plugin:0.7.0',
 
                 netty: 'io.netty:netty-codec-http2:4.1.0.Beta5',
                 netty_tcnative: 'io.netty:netty-tcnative:1.1.33.Fork2' + tcnative_suffix,

